lib/jnpr/junos/utils/fs.py

```diff
@@ -264,6 +265,35 @@ def _decode(fs):
         return dict((_name(fs), _decode(fs)) for fs in rsp.xpath('filesystem'))
 
     # -------------------------------------------------------------------------
+    # directory_usage - filesystem directory usage
+    # -------------------------------------------------------------------------
+
+    def directory_usage(self, path="."):
+        """
+        Returns the directory usage, similar to the unix "du" command.
+
+        :returns: approximate directory usage, including subdirectories, in bytes
```

Comment:
Rather than returning a single integer, I suggest returning a dictionary similar to the return value of `storage_usage()`. The first level key would be directory name. The second level keys could be `blocks`, `bytes`, and `size` (where `size` would be the 'human-readable' value with units).

Returning this type of data structure allows the support of the `depth > 0` situation and allows returning the values in multiple formats so the user can easily access their preferred format without having to perform conversion.